app/src/main/java/com/example/bucketbuds/FriendsAdapter.java

```diff
@@ -0,0 +1,188 @@
+            if (friendsBool) {
+                ivProfileImage = itemView.findViewById(R.id.ivProfileImageIF);
+                tvUsername = itemView.findViewById(R.id.tvUsernameIF);
+                tvFirstName = itemView.findViewById(R.id.tvFirstNameIF);
+                tvLastName = itemView.findViewById(R.id.tvLastNameIF);
+                tvFriendCount = itemView.findViewById(R.id.tvFriendCountIF);
+                tvBucketCount = itemView.findViewById(R.id.tvBucketCountIF);
+                ivRemoveFriend = itemView.findViewById(R.id.ivRemoveFriendIF);
+            } else {
+                ivProfileImage = itemView.findViewById(R.id.ivProfileImageIAF);
+                tvUsername = itemView.findViewById(R.id.tvUsernameIAF);
+                tvFirstName = itemView.findViewById(R.id.tvFirstNameIAF);
+                tvLastName = itemView.findViewById(R.id.tvLastNameIAF);
+                ivAddFriend = itemView.findViewById(R.id.ivAddFriendIAF);
+            }
+        }
+
+        public void bind(User user, Boolean friendsBool){
+            if (friendsBool) {
+                UserPub otherUserPub = user.getUserPub();
+                tvBucketCount.setText(String.valueOf(otherUserPub.getBucketCount()));
+                tvFriendCount.setText(String.valueOf(otherUserPub.getFriendCount()));
+                // user selects a friend to remove
+                View.OnClickListener removeFriendClickListener = new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        otherUserPub.removeFriend(currentUser);
+                        currentUserPub.removeFriend(user);
+                        users.remove(user);
+                        updateFriends(false, getAdapterPosition(), otherUserPub);
+                    }
+                };
+                ivRemoveFriend.setOnClickListener(removeFriendClickListener);
+            } else {
+                // user is already added as a friend but appears in search
+                if (friendsId.contains(user.getObjectId())){
+                    ivAddFriend.setVisibility(View.GONE);
+                } else {
+                    ivAddFriend.setVisibility(View.VISIBLE);
+                    // user selects a friend to add
+                    View.OnClickListener addFriendClickListener = new View.OnClickListener() {
+                        @Override
+                        public void onClick(View v) {
+                            UserPub otherUserPub = user.getUserPub();
+                            otherUserPub.addFriend(currentUser);
+                            currentUserPub.addFriend(user);
+                            ivAddFriend.setVisibility(View.GONE);
+                            updateFriends(true, getAdapterPosition(), otherUserPub);
+                        }
+                    };
+                    ivAddFriend.setOnClickListener(addFriendClickListener);
+                }
+            }
```

Comment:
nit: seems like you are doing a lot of `if(friendsBool) / else`.
You could benefit from some helper methods:
```
// Call this if friendsBool is true
private void bindFriends(User user) {
  UserPub otherUserPub = ....
 ...
}
// Call this if friendsBool is false
private void bindNotFriends(User user) {
  if(friendsId.contains(user.getObjectId()))...
...
}
public void bind(User user, Boolean friendsBool) {
  if(friendsBool) {
    bindFriends(user);
  } else {
    bindNotFriends(user);
  }
```
... You could keep any common logic in the o.g. `bind` method.